rats-devtools/src/python/rats/annotations/_groups.py

```diff
@@ -0,0 +1,13 @@
+from typing import Generic, NamedTuple, TypeVar
+
+T_GroupType = TypeVar("T_GroupType", bound=NamedTuple)
+
+
+class GroupAnnotations(NamedTuple, Generic[T_GroupType]):
```

Comment:
Added some tests for these, and got the following:

```
test/python/rats_test/devtools/test_annotations.py:1: in <module>
    from rats.annotations import FunctionAnnotations, FunctionAnnotationsBuilder, GroupAnnotations
src/python/rats/annotations/__init__.py:7: in <module>
    from ._functions import FunctionAnnotations, FunctionAnnotationsBuilder
src/python/rats/annotations/_functions.py:6: in <module>
    from ._groups import GroupAnnotations, T_GroupType
src/python/rats/annotations/_groups.py:6: in <module>
    class GroupAnnotations(NamedTuple, Generic[T_GroupType]):
/usr/local/lib/python3.10/typing.py:2330: in _namedtuple_mro_entries
    raise TypeError("Multiple inheritance with NamedTuple is not supported")
E   TypeError: Multiple inheritance with NamedTuple is not supported
```

You can use a `dataclass` as an alternative here if you want to keep the `Generic[T_GroupType]`, though

Reply:
the `NamedTuple` in `typing_extensions` supports this so i think i just missed a few but i hope to have gotten to these late last night

Reply:
Will refactor back to NamedTuple in my branch, too